File: src/main/service/Handler/rc_handler.php
<?php

require_once '../../model/rice_cooker.php';

class RcHandler {
    private static $rc_list = [];

    public static function add($id, $is_free) {
        $new_rc = new RiceCooker($id, $is_free);
        self::$rc_list[] = $new_rc;
        echo "New rice cooker with id: {$id} successfully added." . PHP_EOL;
    }

    public static function change_state($id, $target_attr, $state) {
        $target_rc = self::get_rc($id);
        if ($target_rc) {
            switch ($target_attr) {
                case 'is_free':
                    $target_rc->setIsFree($state);
                    break;
                case 'is_cooking':
                    $target_rc->setIsCooking($state);
                    break;
                case 'is_plugged':
                    $target_rc->setIsPlugged($state);
                    break;
                default:
                    echo 'The targeted attribute is not valid' . PHP_EOL;
            }
        } else {
            echo "The rice cooker with id: {$id} doesn't exist." . PHP_EOL;
        }
    }

    public static function rc_list() {
        return self::$rc_list;
    }

    public static function get_rc($id) {
        $found = array_filter(self::$rc_list, function ($rc) use ($id) {
            return $rc->getId() == $id;
        });

        if (!empty($found)) {
            return reset($found);
        } else {
            echo "The rice cooker with id: {$id} doesn't exist." . PHP_EOL;
        }
    }
}

?>

File: src/main/service/Exception/rc_handler.rb
require_relative '../../model/rice_cooker'

module RcHandler
  @rc_list = []

  def self.add(id, is_free)
    new_rc = RiceCooker.new(id, is_free)
    @rc_list.push(new_rc)
    puts "New rice cooker with id: #{id} successfully added."
  end

  def self.change_state(id, target_attr, state)
    target_rc = @rc_list.find { |rc| rc.id == id }
    if target_rc
      case target_attr
      when 'is_free'
        target_rc.is_free = state
      when 'is_cooking'
        target_rc.is_cooking = state
      when 'is_plugged'
        target_rc.is_plugged = state
      else
        puts 'The targeted attribute is not valid'
      end
    else
      puts "The rice cooker with id: #{id} doesn't exist."
    end
  end

  def self.rc_list
    @rc_list
  end

  def self.get_rc(id)
    found = @rc_list.find { |rc| rc.id == id }
    return found if found

    puts "The rice cooker with id: #{id} doesn't exist."
  end
end
